src/explosion.cpp

```diff
@@ -321,7 +321,9 @@ void game::explosion( const tripoint &p, const explosion_data &ex )
             int qty = shr.casing_mass * std::min( 1.0, shr.recovery / 100.0 ) /
                       to_gram( fragment_drop->weight );
             // Truncate to a random selection
-            std::random_shuffle( tiles.begin(), tiles.end() );
+            static auto eng = std::default_random_engine(
+                                  std::chrono::system_clock::now().time_since_epoch().count() );
+            std::shuffle( tiles.begin(), tiles.end(), eng );
```

Comment:
This change is going to make explosions less reproducible, which might lead to some annoying-to-pin-down test failures.  But there's already other `shuffle` calls with the same issue; probably makes sense to fix them all at once.

Reply:
I've just changed `std::random_shuffle` to `std:shuffle` everywhere. That was the last unchanged instance.

Do you think we need to change random engine seed from system clock to something else ?